### PR TITLE
[hive-source] add source connector for hive

### PR DESCRIPTION
**Pull request by winar-jin**
## What
Add source connector for Apache Hive to load data.

### CODE REVIEW

**Comment by sherifnada** [comment]
@winar-jin thanks for getting started on the Hive source! Please let us know if you have any questions or feedback. We're more than happy to help however we can :)

**Review by jrhizor — CHANGES_REQUESTED** [review]
Thanks for the PR @winar-jin !

Have you thought about using the Hive JDBC driver instead? We have a few Java JBDC-based drivers that re-use most of the core integration code. Here's MySQL for example: [source-mysql](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-mysql/) which mostly boils down to [defining the spec](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mysql/src/main/resources/spec.json) and [mapping that to a JDBC connect config](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java). Almost all of the core logic (which seems to resemble what you're doing in this PR) lives in [AbstractJdbcSource](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java). Are there any limitations to using JDBC for Hive?

 Let me know if you have any questions, I'm happy to discuss here or on Slack.

**Comment by winar-jin** [comment]
Thanks for the detailed review and the comment. I will check all the comments and make changes to this pull request.  @jrhizor

File: airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/d78dda21b4ec4f51a9f220718b94b844.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "sourceDefinitionId": "d78dda21b4ec4f51a9f220718b94b844",
```
[the remark this answers is not in the document]
Comment (winar-jin) [reply airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/d78dda21b4ec4f51a9f220718b94b844.json:2]: Oops, I just use `python -c 'import sys,uuid; sys.stdout.write(uuid.uuid4().hex)' | pbcopy && pbpaste && echo` to get the UUID from the terminal, I thought it was valid. But I should have to check it, sorry for the mistake.

File: airbyte-integrations/connectors/source-hive/source_hive/source.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            logger.info("Hive Source: Start to connect to Hive server ......")
+            conn, cur = connect_to_hive(json_config)
+            cur.execute("SELECT VERSION()")
+            version = cur.fetchone()
+            logger.info("Hive Source: Connect successful! The Hive version is " + str(version))
+
+        except Exception as e:
+            logger.error("Hive Source: Connect failed! The reason is " + str(e))
+            return AirbyteConnectionStatus(status=Status.FAILED)
+        cur.close()
```
[the remark this answers is not in the document]
Comment (winar-jin) [reply airbyte-integrations/connectors/source-hive/source_hive/source.py:64]: Sorry, I encountered obstacles in solving this problem. Seems like the with statement works with a class which implemented `__enter__` and `__exit__` method, but for this case, it's `connect_to_hive` method, I didn't work it out to use with the statement. Sorry for that, I just add finally to wrap the close sentence.

**Comment by winar-jin** [comment]
> Have you thought about using the Hive JDBC driver instead? We have a few Java JBDC-based drivers that re-use most of the core integration code. Here's MySQL for example: [source-mysql](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-mysql/) which mostly boils down to [defining the spec](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mysql/src/main/resources/spec.json) and [mapping that to a JDBC connect config](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java). Almost all of the core logic (which seems to resemble what you're doing in this PR) lives in [AbstractJdbcSource](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java). Are there any limitations to using JDBC for Hive?

I think Hive should support to connect it by using JDBC  driver. And I think under the hook of `impala` dependency it may use the JDBC driver to connect Hive.

But for this suggestion, Do you mean that I can just use a Java source, which is similar to `MySqlSource.java` file to do this job? 

And when I use `npm run generate` command to generate a source, I think there doesn't have an option for Java source, just Python source, Java destination, and Singer tap. If I want to have a Java source, should I choose a Java destination to do this job? 

@jrhizor  Thanks sincerely! And waiting for your warmly reply. 🎁

**Comment by jrhizor** [comment]
> But for this suggestion, Do you mean that I can just use a Java source, which is similar to MySqlSource.java file to do this job?
Yes, I think that'd be possible.

We don't have a generator based on templates for the JDBC sources yet, but it should be possible to copy `source-mysql` or `source-postgres` into a new directory. There are way fewer files involved in the JDBC sources.

- `build.gradle` - needs to add a dependency on the `org.apache.hive:hive-jdbc:3.1.2` driver and specify the correct `mainClass` 
- `Dockerfile` - would need to set `io.airbyte.name=airbyte/source-hive` and `ENV APPLICATION source-hive`
- `spec.json` - the location is different but the contents would be the same as they are in this PR
- The equivalent of `MySqlSource` needs to specify the correct Hive JDBC driver class and a method of converting your spec's config to a JDBC config
- the rest of the files are testing related

Are you interested in trying out that approach?

**Comment by jrhizor** [comment]
This looks good! Have you had a chance to test it on an actual Hive source to see if it can replicate data?

**Comment by jrhizor** [comment]
Hey @winar-jin, just wanted to check in and ask if you needed any help testing this against a Hive server. Please let me know if you need anything.

**Comment by winar-jin** [comment]
Hi @jrhizor , Sorry for the late reply and late verify of these codes. I just join a new launched project these two weeks, cause of the late work almost everyday, I missed some code work of AirByte. Again, I am sorry for that. I will pick up this work as soon as I could, and I will call for help as I need it. Thanks.

**Pull request closed without merging**